### ISSUE TITLE
Scope of project

### ISSUE THREAD

**Issue by LecrisUT** [fedora-infra/anitya#1797]
I think it would be nice to have some overview of the scope of the project. For example retrieving project versions and offering API endpoints to query are clearly within the scope, but things like offering configurable webhooks for distros is a bit foggy.

This can be very vague so I try to help with a few points that could help steer the discussion. I don't believe most things that I write as potential should be within scope, but I try to add them for completeness:

#### Ways to broadcast the found versions
 **Currently:** rest API, fedora-messaging
 **Potential:** distro-specific webhooks, RSS-feed, other web API like graphql

#### Ways to retrieve versions
**Currently:** quite comprehensive
**Potential:** more backends if PRs are made?

#### Release information
**Currently:** version, project definition
**Potential:** release notes, timestamp of release, reason for drop release, contributors list, authors, source changes, license changes, release type, security updates

#### Filtering and transforming versions
**Currently:**  filter latest and stable versions, filter tags by semver/calendar etc.
**Potential:** customizable regex transformations, more filters like newest version, regex pattern, etc.

#### Authorization
**Currently:** logged-in users can do anything
**Potential:** limit editing of distro-options (currently none) to admins/proven-packagers

#### Authentication
**Currently:** FAS account
**Potential:** Other OAuth linked to supported backends (Github, Gitlab, etc.), ephemeral users (not stored locally)

#### Project/distro management
**Currently:** 
**Potential:** 

#### Code structure
Modernizing the python project, changing the build backend, organizing files, `src-layout` or not, etc.

**Comment by Zlopez** [comment]
That is a good summary, I just want to add that for OAuth support there is a [PR](https://github.com/fedora-infra/anitya/pull/1220) open right now.

And there are plenty of other open issues that are addressing the potential. The only problem I have is to find the time to work on them.

**Comment by LecrisUT** [comment]
Indeed, I was trying more to get an overview of what needs prioritization, what we should deffer to other projects, what needs help, which topics are difficult to research/could use discussions from community/contributors, etc.
